### PR TITLE
Container factory method

### LINKED ISSUE #28: create handler dynamically

### PR DESCRIPTION
**Pull request by darlison-alves**
Container class created to manage "Handler class" and dependency injection in the same.
How exemple, I writed some tests....

### CODE REVIEW

**Comment by cl3dson** [comment]
Many new issues in codify code quality review, please could you check some of them?

**Comment by iagosrodrigues** [comment]
You could just see in the details [here](https://app.codacy.com/manual/f.cl3dson/mercury/pullRequest?prid=5070860) and fix the issues.

Before the merge, please squash them up or use a fixup. This is required to remove the "fix" commits that don't have helpful messages.

**Review by iagosrodrigues — CHANGES_REQUESTED** [review]
Have to fix the Codacy issues before merge.

**Comment by iagosrodrigues** [comment]
I'll see this at the weekend.

File: lib/connection/RabbitMQConnectionFacade.ts [hunk 1]
```diff
@@ -153,14 +157,17 @@ export class RabbitMQConnectionFacade {
     }
 
     public async dispatchMessage(descriptor: string, msg: ConsumeMessage): Promise<void> {
-        const handlers = Mercury.handlerRegistry;
+        //const handlers = Mercury.handlerRegistry;
+
```
Comment (iagosrodrigues) [inline lib/connection/RabbitMQConnectionFacade.ts:161]: I think you should remove the commented code as well.
```suggestion
```

File: lib/connection/RabbitMQConnectionFacade.ts [hunk 2]
```diff
@@ -153,14 +157,17 @@ export class RabbitMQConnectionFacade {
...
 
         if (Reflect.hasMetadata('messageBindings', Mercury.prototype.constructor)) {
             const bindings: Map<string, string> = Reflect.getMetadata('messageBindings', Mercury.prototype.constructor);
             const handlerClassName = bindings.get(descriptor);
             if (handlerClassName) {
-                if (handlers.has(handlerClassName)) {
-                    const handler = handlers.get(handlerClassName);
+                //if (handlers.has(handlerClassName))
+                if (handler) {
+                    //const handler = handlers.get(handlerClassName);
```
Comment (iagosrodrigues) [inline lib/connection/RabbitMQConnectionFacade.ts:170]: Here too.
```suggestion
                if (handler) {
```

**Review by iagosrodrigues — CHANGES_REQUESTED** [review]
Here more tips.

**Pull request merged**